subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java

```diff
@@ -117,6 +117,7 @@ private static void configureScalaCompile(final Project project, final SourceSet
 
         Convention scalaConvention = (Convention) InvokerHelper.getProperty(sourceSet, "convention");
         ScalaSourceSet scalaSourceSet = scalaConvention.findPlugin(ScalaSourceSet.class);
+        SourceSetUtil.configureOutputDirectoryForSourceSet(sourceSet, scalaSourceSet.getScala(), project);
         SourceSetUtil.configureForSourceSet(sourceSet, scalaSourceSet.getScala(), scalaCompile, scalaCompile.getOptions(), project);
```

Comment:
It doesn't particularly make a difference to call it before, it's more the fact of calling it outside a lazy task configuration. Since the configuration in `configureOutputDirectoryForSourceSet` is independent of the task calling it from inside `configureForSourceSet` is wrong. If the task doesn't get realized, it won't configure the output directory properly.

Reply:
Makes sense. Thanks for the explanation! 👍